### PR TITLE
Moved copy-by-click function into mmui-web.

### PR DESCRIPTION
**Pull request by kanastasakis26**

### CODE REVIEW

File: src/ts/mmui-widget/mmui-copy.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        buffer.style.left = '-99999px';
+
+        document.body.appendChild(buffer);
+
+        buffer.select();
+        document.execCommand('copy');
+
+        document.body.removeChild(buffer);
+
+        $(this).popover(
```
Comment (besseym) [inline src/ts/mmui-widget/mmui-copy.ts]: Perhaps we should separate the copy functionality from the popover. We may want to do a copy with out a popover.

Comment (kanastasakis26) [reply src/ts/mmui-widget/mmui-copy.ts]: I will make it conditional.

File: src/ts/mmui-widget/mmui-copy.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ *           trigger.click(createCopyClickHandler(text));
+ *       }
+ *   );
+ *
+ * @param text The text to be copied.
+ * @param popover Boolean that indicates whether a popover is used as feedback.
+ * @param timeout The time in millisecond =s to display feedback popover.
+ * @returns {Function} A handler function.
+ */
+export function createCopyClickHandler(text, popover=true, timeout=750){
```
Comment (besseym) [inline src/ts/mmui-widget/mmui-copy.ts]: We could also make the content of the popup configurable as well.

File: src/ts/mmui-widget/mmui-copy.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    return function (){
```
Comment (besseym) [inline src/ts/mmui-widget/mmui-copy.ts]: I'm thinking that the text should actually be passed into the retuned click handler function. The text may change after the handler function has been applied.

File: src/ts/mmui-widget/mmui-copy.ts (hunk 2 above)
Comment (besseym) [reply src/ts/mmui-widget/mmui-copy.ts]: Perhaps we should just call it createCopyHandler. The copy action maybe initiated by an event that isn't a click.

File: src/ts/mmui-widget/mmui-copy.ts (hunk 3 above)
Comment (kanastasakis26) [reply src/ts/mmui-widget/mmui-copy.ts]: Not sure what you mean here. I moved the code that does the copying into a separate function. If someone wants to do custom stuff, now they can.

**Pull request merged**